queue.go

```diff
@@ -22,6 +24,14 @@ var (
 	publisherWid uint64 = 0
 )
 
+type url struct {
```

Comment:
please export it by renaming to `URL`:
![image](https://user-images.githubusercontent.com/11900380/135455972-10eb3bd9-973f-42c5-80a1-8320027f91da.png)

Reply:
i have done it but

but why do we want to make it public